File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/Task.java
package com.roxiemobile.networkingapi.network.rest;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface Task<Ti, To> extends Call<Ti>, Cloneable {

    /**
     * TODO
     */
    void execute(@Nullable Callback<Ti, To> callback);

    /**
     * TODO
     */
    @NotNull Cancellable enqueue(@Nullable Callback<Ti, To> callback, boolean callbackOnUiThread);

    /**
     * TODO
     */
    @NotNull Task<Ti, To> clone();
}

File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/Task.kt
package com.roxiemobile.networkingapi.network.rest

interface Task<Ti, To>: Call<Ti>, Cloneable {

// MARK: - Methods

    /**
     * TODO
     */
    fun execute(callback: Callback<Ti, To>?)

    /**
     * TODO
     */
    fun enqueue(callback: Callback<Ti, To>?, callbackOnUiThread: Boolean): Cancellable

    /**
     * TODO
     */
    public override fun clone(): Task<Ti, To>
}
